nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -0,0 +1,201 @@
+            commentModel = s.createDocument(commentModel);
+            return commentModel.getId();
+        });
+    }
+
+    @Override
+    public Comment getComment(CoreSession session, String commentId) {
+        return CoreInstance.doPrivileged(session, s -> {
+            DocumentModel commentModel = s.getDocument(new IdRef(commentId));
+            if (commentModel == null) {
+                if (log.isWarnEnabled()) {
+                    log.warn("The comment " + commentId + " does not exist.");
+                }
+                return null;
+            }
+            Comment comment = new CommentImpl();
+            comment.setAuthor((String) commentModel.getPropertyValue(COMMENT_AUTHOR));
+            comment.setText((String) commentModel.getPropertyValue(COMMENT_TEXT));
+            comment.setDocumentId((String) commentModel.getPropertyValue(COMMENT_DOCUMENT_ID));
+            comment.setCreationDate((Instant) commentModel.getPropertyValue(COMMENT_CREATION_DATE));
+            comment.setModificationDate((Instant) commentModel.getPropertyValue(COMMENT_MODIFICATION_DATE));
+            return comment;
+        });
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<Comment> getComments(CoreSession session, String documentId) {
+        if (!session.exists(new IdRef(documentId))) {
+            throw new IllegalArgumentException("The document " + documentId + " does not exist.");
+        }
+        return CoreInstance.doPrivileged(session, s -> {
+            PageProviderService ppService = Framework.getService(PageProviderService.class);
+            Map<String, Serializable> props = Collections.singletonMap(
+                    CoreQueryAndFetchPageProvider.CORE_SESSION_PROPERTY, (Serializable) s);
+            List<DocumentModel> commentList = ((PageProvider<DocumentModel>) ppService.getPageProvider(
+                    GET_COMMENTS_FOR_DOC_PAGEPROVIDER_NAME, null, null, null, props, documentId)).getCurrentPage();
+            return commentList.stream().map(commentModel -> {
+                Comment comment = new CommentImpl();
+                comment.setAuthor((String) commentModel.getPropertyValue(COMMENT_AUTHOR));
+                comment.setText((String) commentModel.getPropertyValue(COMMENT_TEXT));
+                comment.setDocumentId(documentId);
+                comment.setCreationDate((Instant) commentModel.getPropertyValue(COMMENT_CREATION_DATE));
+                comment.setModificationDate((Instant) commentModel.getPropertyValue(COMMENT_MODIFICATION_DATE));
+                return comment;
+            }).collect(Collectors.toList());
+        });
+    }
+
+    @Override
+    public void updateComment(CoreSession session, String commentId, Comment comment) {
+        CoreInstance.doPrivileged(session, s -> {
+            DocumentModel commentModel = s.getDocument(new IdRef(commentId));
```

Comment:
Please also do an existence check and throw `IllegalArgumentException` here if the comment does not exist (and add `throws IllegalArgumentException` to the signature